deployment/common/changeset/state.go

```diff
@@ -79,78 +79,90 @@ func MaybeLoadMCMSWithTimelockState(env deployment.Environment, chainSelectors [
 
 // MaybeLoadMCMSWithTimelockChainState looks for the addresses corresponding to
 // contracts deployed with DeployMCMSWithTimelock and loads them into a
-// MCMSWithTimelockState struct. If none of the contracts are found, the state struct will be nil.
+// MCMSWithTimelockState struct.  If none of the contracts are found, the state struct will be nil.
+//
 // An error indicates:
 // - Found but was unable to load a contract
 // - It only found part of the bundle of contracts
 // - If found more than one instance of a contract (we expect one bundle in the given addresses)
-func MaybeLoadMCMSWithTimelockChainState(chain deployment.Chain, addresses map[string]deployment.TypeAndVersion) (*MCMSWithTimelockState, error) {
-	state := MCMSWithTimelockState{
-		MCMSWithTimelockContracts: &proposalutils.MCMSWithTimelockContracts{},
-	}
-	// We expect one of each contract on the chain.
-	timelock := deployment.NewTypeAndVersion(types.RBACTimelock, deployment.Version1_0_0)
-	callProxy := deployment.NewTypeAndVersion(types.CallProxy, deployment.Version1_0_0)
-	proposer := deployment.NewTypeAndVersion(types.ProposerManyChainMultisig, deployment.Version1_0_0)
-	canceller := deployment.NewTypeAndVersion(types.CancellerManyChainMultisig, deployment.Version1_0_0)
-	bypasser := deployment.NewTypeAndVersion(types.BypasserManyChainMultisig, deployment.Version1_0_0)
-	// the same contract can have different roles
-	multichain := deployment.NewTypeAndVersion(types.ManyChainMultisig, deployment.Version1_0_0)
+func MaybeLoadMCMSWithTimelockChainState(
+	chain deployment.Chain,
+	addresses map[string]deployment.TypeAndVersion,
+) (*MCMSWithTimelockState, error) {
+	var (
+		state = MCMSWithTimelockState{
+			MCMSWithTimelockContracts: &proposalutils.MCMSWithTimelockContracts{},
+		}
 
-	// Convert map keys to a slice
-	wantTypes := []deployment.TypeAndVersion{timelock, proposer, canceller, bypasser, callProxy}
+		// We expect one of each contract on the chain.
+		timelock  = deployment.NewTypeAndVersion(types.RBACTimelock, deployment.Version1_0_0)
+		callProxy = deployment.NewTypeAndVersion(types.CallProxy, deployment.Version1_0_0)
+		proposer  = deployment.NewTypeAndVersion(types.ProposerManyChainMultisig, deployment.Version1_0_0)
+		canceller = deployment.NewTypeAndVersion(types.CancellerManyChainMultisig, deployment.Version1_0_0)
+		bypasser  = deployment.NewTypeAndVersion(types.BypasserManyChainMultisig, deployment.Version1_0_0)
 
-	// Ensure we either have the bundle or not.
-	_, err := deployment.AddressesContainBundle(addresses, wantTypes)
+		// the same contract can have different roles
+		multichain = deployment.NewTypeAndVersion(types.ManyChainMultisig, deployment.Version1_0_0)
+
+		// Convert map keys to a slice
+		wantTypes = []deployment.TypeAndVersion{timelock, proposer, canceller, bypasser, callProxy}
```

Comment:
I think we need to update this to include the logic -
1. Either it should have types - proposer, canceller, bypasser,
2. Or it should have type- multichain with different roles

Reply:
I just removed this, it's not clear why we are asserting which contracts are in the address map.  the docstring says that we should expect nil state values.